ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -107,18 +106,17 @@ public AggregatingAttestationPool(
   }
 
   public synchronized void add(final ValidatableAttestation attestation) {
-    final AttestationData attestationData = attestation.getAttestation().getData();
-    final Supplier<Int2IntMap> committeesSizeSupplier =
-        attestation
-            .getCommitteesSize()
-            .<Supplier<Int2IntMap>>map(committeesSize -> () -> committeesSize)
-            .orElseGet(() -> getCommitteesSizeSupplierUsingTheState(attestationData));
-    final boolean add =
-        getOrCreateAttestationGroup(attestationData, committeesSizeSupplier).add(attestation);
-    if (add) {
-      updateSize(1);
-    }
-    // Always keep the latest slot attestations so we don't discard everything
+    final Optional<Int2IntMap> committeesSize =
+        attestation.getCommitteesSize().or(() -> getCommitteesSize(attestation.getAttestation()));
+    getOrCreateAttestationGroup(attestation.getAttestation(), committeesSize)
+        .ifPresent(
+            attestationGroup -> {
+              final boolean add = attestationGroup.add(attestation);
+              if (add) {
```

Comment:
mm i get what you're doing, but i wonder if either 'added' or 'success' or just calling the add in the if fn would be cleaner... `add` is almost right but may be cleaner in another way to read.